kinto_wizard/yaml2kinto.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function
+from .logger import logger
+from .kinto2yaml import introspect_server
+
+
+def initialize_server(client, config):
+    logger.debug("Converting YAML config into a server batch.")
+    # 1. Introspect current server state.
+    current_server_status = introspect_server(client)
+    # 2. For each bucket
+    with client.batch() as batch:
+        for bucket_id, bucket in config.items():
+            bucket_exists = bucket_id in current_server_status
+            bucket_data = bucket.get('data', {})
+            bucket_permissions = bucket.get('permissions', {})
+            bucket_groups = bucket.get('groups', {})
+            bucket_collections = bucket.get('collections', {})
+
+            if not bucket_exists:
+                bucket_current_groups = {}
+                bucket_current_collections = {}
+
+                # Create the bucket if not present in the introspection
+                batch.create_bucket(bucket_id,
+                                    data=bucket_data,
+                                    permissions=bucket_permissions)
+            else:
+                bucket_current_groups = current_server_status[bucket_id]['groups']
+                bucket_current_collections = current_server_status[bucket_id]['collections']
+
+                # Patch the bucket if mandatory
+                current_bucket = current_server_status[bucket_id]
+                current_bucket_data = current_bucket.get('data', {})
+                current_bucket_permissions = current_bucket.get('permissions', {})
+
+                if (current_bucket_data != bucket_data or
+                        current_bucket_permissions != bucket_permissions):
+                    batch.patch_bucket(bucket_id,
+                                       data=bucket_data,
+                                       permissions=bucket_permissions)
+
+            # 1.1 For each group, patch it if needed
+            for group_id, group_info in bucket_groups.items():
+                group_exists = bucket_exists and group_id in bucket_current_groups
```

Comment:
You don't really need to check `bucket_exists` here (although I guess it doesn't hurt).